Apps/W1/EDocument/app/src/Processing/Import/StructureReceivedEDocument/EDocumentPEPPOLHandler.Codeunit.al

```diff
@@ -81,126 +85,152 @@ codeunit 6173 "E-Document PEPPOL Handler" implements IStructuredFormatReader
-        SetNumberValueInField(LineXML, XMLNamespaces, 'cac:InvoiceLine/cac:Price/cbc:PriceAmount', EDocumentPurchaseLine."Unit Price");
-    end;
-
-    local procedure SetCurrencyValueInField(XMLDocument: XmlDocument; XMLNamespaces: XmlNamespaceManager; Path: Text; MaxLength: Integer; var CurrencyField: Code[10])
-    var
-        GLSetup: Record "General Ledger Setup";
-        XMLNode: XmlNode;
-        CurrencyCode: Code[10];
-    begin
-        if not XMLDocument.SelectSingleNode(Path, XMLNamespaces, XMLNode) then
-            exit;
-
-        GLSetup.Get();
-
-        if XMLNode.IsXmlElement() then begin
-            CurrencyCode := CopyStr(XMLNode.AsXmlElement().InnerText(), 1, MaxLength);
-            if GLSetup."LCY Code" <> CurrencyCode then
-                CurrencyField := CurrencyCode;
-            exit;
-        end;
-
-        if XMLNode.IsXmlAttribute() then begin
-            CurrencyCode := CopyStr(XMLNode.AsXmlAttribute().Value, 1, MaxLength);
-            if GLSetup."LCY Code" <> CurrencyCode then
-                CurrencyField := CurrencyCode;
-            exit;
-        end;
+                EDocumentXMLHelper.SetStringValueInField(PeppolXML, XmlNamespaces, '/inv:Invoice/cac:AccountingSupplierParty/cac:Party/cbc:EndpointID', MaxStrLen(EDocumentPurchaseHeader."Vendor GLN"), EDocumentPurchaseHeader."Vendor GLN");
     end;
-#pragma warning restore AA0139
 
-    local procedure SetStringValueInField(XMLDocument: XmlDocument; XMLNamespaces: XmlNamespaceManager; Path: Text; MaxLength: Integer; var Field: Text)
+    local procedure PopulatePurchaseCreditMemoHeader(PeppolXML: XmlDocument; XmlNamespaces: XmlNamespaceManager; var EDocumentPurchaseHeader: Record "E-Document Purchase Header")
     var
+        EDocumentXMLHelper: Codeunit "EDocument XML Helper";
         XMLNode: XmlNode;
+        CreditNoteNamespaceLbl: Label 'urn:oasis:names:specification:ubl:schema:xsd:CreditNote-2';
     begin
-        if not XMLDocument.SelectSingleNode(Path, XMLNamespaces, XMLNode) then
-            exit;
+        // Add CreditNote namespace for proper header parsing
+        XmlNamespaces.AddNamespace('cn', CreditNoteNamespaceLbl);
 
-        if XMLNode.IsXmlElement() then begin
-            Field := CopyStr(XMLNode.AsXmlElement().InnerText(), 1, MaxLength);
-            exit;
-        end;
-
-        if XMLNode.IsXmlAttribute() then begin
-            Field := CopyStr(XMLNode.AsXmlAttribute().Value(), 1, MaxLength);
-            exit;
-        end;
+        EDocumentXMLHelper.SetStringValueInField(PeppolXML, XMLNamespaces, '/cn:CreditNote/cbc:ID', MaxStrLen(EDocumentPurchaseHeader."Sales Invoice No."), EDocumentPurchaseHeader."Sales Invoice No.");
+        EDocumentXMLHelper.SetStringValueInField(PeppolXML, XMLNamespaces, '/cn:CreditNote/cac:BillingReference/cac:InvoiceDocumentReference/cbc:ID', MaxStrLen(EDocumentPurchaseHeader."Purchase Order No."), EDocumentPurchaseHeader."Purchase Order No.");
```

Comment:
This is not correct. 
This refers to a invoice in our system, that the credit memo applied to. 
We actually dont have such a field on the edoc purch header, but we should have so we can read this number into it.